### ISSUE TITLE
Add ability to move tasks to top of sections

### ISSUE THREAD

**Issue by dencold** [dencold/attainment-web#60]
Right now our task sections are treated as a static list. It would be nice if we could decide what is at the top of the list as the highest priority item, etc.

**Comment by dencold** [comment]
I'd like to use the `;` key to quickly jump to the top, and we can use `'` to drop to the bottom.

**Comment by dencold** [comment]
This is going to be a decently hard but of work in implementation. The current data structure just keeps one list of tasks and the client does the work to filter and sort on date. 

We'll need to figure out the best way to store this on the backend

**Comment by dencold** [comment]
Thought about this some more and it got even harder. Because we use firebase's cloud datastore for attainment, we don't get the luxury of relationships and you start to create mapping objects to represent these. So I'd likely need to have a new entity for every place we'd want to enforce an user-specified order. We'd need to have something that represents the order for the Inbox and the Weekend View and the Today view, etc. etc.

That's not too big a deal, we'd just have a list of taskIds to keep track of this and it's easy to store this in firebase. The hard part comes with keeping all these things in sync. If you switched something from the Inbox to Today, for example, you'd update the state on the task AND then delete the taskId from the array in Inbox and add the taskId to the Today array. It gets more complicated when you have projects in the mix as well. 

I wanted to avoid making a lot of these new data structures and logic for syncing, so I've found a workaround that should work:

90% of what I'm looking for is just getting the most important task to the top of the list. I decided to take advantage of the `isStarred` field to make this work and actually give a better user experience to this as well. Now when you star a task it will go up to the top of whatever list your in. When you unstar it it will stay at the top of the list or below other stars if they exist.
